src/tools/miri/src/bin/miri.rs

```diff
@@ -329,6 +329,13 @@ impl rustc_driver::Callbacks for MiriBeRustCompilerCalls {
 }
 
 fn exit(exit_code: i32) -> ! {
+    // MIR validation uses delayed bugs. Flush them as we won't return for `run_compiler` to do it.
+    ty::tls::with_opt(|opt_tcx| {
+        if let Some(tcx) = opt_tcx {
+            tcx.dcx().flush_delayed()
```

Comment:
Seems a bit odd to only do this at the end... I assume this is fairly cheap? Maybe we should just do it in `find_mir_or_eval_fn` after `load_mir`?